packages/replayio/src/commands/record.ts

```diff
@@ -9,44 +9,79 @@ import { printRecordings } from "../utils/recordings/printRecordings";
 import { selectRecordings } from "../utils/recordings/selectRecordings";
 import { LocalRecording } from "../utils/recordings/types";
 import { uploadRecordings } from "../utils/recordings/upload/uploadRecordings";
+import { logPromise } from "../utils/async/logPromise";
 
 registerCommand("record", { checkForRuntimeUpdate: true, requireAuthentication: true })
   .argument("[url]", `URL to open (default: "about:blank")`)
   .description("Launch the replay browser in recording mode")
   .action(record);
 
 async function record(url: string = "about:blank") {
-  const recordingsBefore = await getRecordings();
+  const prevRecordings = await getRecordings();
 
   await launchBrowser(url, { processGroupId: uuid() });
 
   const recordingsAfter = await getRecordings();
-  const recordingsNew = recordingsAfter.filter(
-    recording => !recordingsBefore.some(({ id }) => id === recording.id)
-  );
+
+  const nextCrashedRecordings: LocalRecording[] = [];
+  const nextRecordings: LocalRecording[] = [];
+
+  recordingsAfter.filter(recording => {
+    if (!prevRecordings.some(({ id }) => id === recording.id)) {
+      if (recording.recordingStatus === "crashed") {
+        nextCrashedRecordings.push(recording);
+      } else {
+        nextRecordings.push(recording);
+      }
+    }
+  });
 
   console.log(""); // Spacing for readability
 
-  if (recordingsNew.length > 0) {
+  // First check for any new crashes; these we should upload automatically
+  if (nextCrashedRecordings.length > 0) {
+    console.log(
+      "It looks like something went wrong with this recording. Please hold while we upload crash data."
+    );
+
+    const promise = uploadRecordings(nextCrashedRecordings, {
+      processAfterUpload: false,
+      silent: true,
+    });
+
+    logPromise(promise, {
+      messages: {
+        pending: "Uploading crash data...",
+        success: "Crash data uploaded successfully",
```

Comment:
if you intentionally dont want to render anything here when it fails then maybe a comment here would be helpful explianing? it's tempting to add that error message here

Reply:
I thought about that but I'm not sure what we would tell the user.

At that point, the runtime has crashed _and_ we failed to upload crash data. It looks pretty clowny and there isn't any action the user take (beyond maybe reporting this to us via a ticket). Which maybe that is what we should suggest, I dunno.

Reply:
Going to land this as-is but happy to iterate on the messaging if you have suggestions.

Reply:
> It looks pretty clowny and there isn't any action the user take (beyond maybe reporting this to us via a ticket).

Even if the user can't take any action in this situation, I think it's still reasonable to let them know what happened. It's about expectations. Like, they would see that we were uploading the crashed data and they would assume that it went well if we won't tell them otherwise. So they could come back to us after a while to ask if the issue got resolved or something. At that point, we'd try to find the data associated with that user and we wouldn't find any. It's hypothetical and I don't know if it's even possible to look for those crashes in such a way - but maybe it will be in the future? Either way, being honest about it here could save both parties from wasting some time on assuming that the operation has succeeded.